src/lib/chawathe98_matcher.rs

```diff
@@ -619,6 +696,580 @@ impl<T: Clone + Debug + Eq + 'static> MappingStoreGraph<T> {
+                                curr_edges = Some((key, to_node, *edge_type, edge_cost));
+                                min_edge_value = edge_cost;
+                            } else {
+                                all_removed_edges.insert((key, to_node, *edge_type, edge_cost));
+                            }
+                        } else if min_edge_value < current_edge.3 {
+                            all_removed_edges.insert(current_edge);
+                            curr_edges = Some((key, to_node, *edge_type, edge_cost));
+                            min_edge_value = edge_cost;
+                        } else {
+                            all_removed_edges.insert((key, to_node, *edge_type, edge_cost));
+                        }
+                    }
+                }
+            }
+            all_remaining_edges.insert(curr_edges.unwrap());
+
+
+        } else {
+            for &(to_node, ref edge_type1, edge_cost) in &val.0.clone(){
+                let edge_type = edge_type1;
+                all_remaining_edges.insert((key,to_node,*edge_type,edge_cost));
+            }
+        }
+    }
+
+    /*
+    All the removed edges should have their own replacement.
+    */
+    // Insert Node
+    let last_node_from_node_insert = NodeId::<FromNodeId>::new(new_matcher_pruning.from_arena.borrow().size() - 1);
+    // Delete Node
+    let last_node_to_node_delete = NodeId::<ToNodeId>::new(new_matcher_pruning.to_arena.borrow().size() - 1);
+
+    for &(from,to,edge_type,cost) in &all_removed_edges{
+        if edge_type == EdgeType::MOVE {
+            // Direct insert.
+            if !all_remaining_edges.contains(&(last_node_from_node_insert, to, EdgeType::INSERT,cost)){
+                all_remaining_edges.insert((last_node_from_node_insert, to, EdgeType::INSERT,cost));
+            }
+        } else if edge_type == EdgeType::GLUE {
+            let all_descendants = to.descendants(&new_matcher_pruning.to_arena.borrow()).collect::<Vec<NodeId<ToNodeId>>>();
+            if !all_remaining_edges.contains(&(last_node_from_node_insert, to, EdgeType::INSERT,cost)){
+                all_remaining_edges.insert((last_node_from_node_insert, to, EdgeType::INSERT,cost));
+            }
+            for to_node_descendants in all_descendants {
+                if !all_remaining_edges.contains(&(last_node_from_node_insert, to_node_descendants, EdgeType::INSERT,cost)){
+                    all_remaining_edges.insert((last_node_from_node_insert, to_node_descendants, EdgeType::INSERT,cost));
+                }
+            }
+        }
+    }
+
+
+    // Create new edges
+    let mut vec_edges:Vec<Edge> = Vec::new();
+    for (from,to,edge_type,cost) in all_remaining_edges{
+        let edge = Edge::new(from,to,cost,edge_type);
+        vec_edges.push(edge);
+    }
+
+    new_matcher_pruning.list_edges = RefCell::new(vec_edges);
+
+
+    let file_location = "C:/Users/Abhishek Shankar/Desktop/Rust Repost/diffract/test_AS_6_edge_final.dot";
```

Comment:
Eek! Hard-coded filename -- that can't be right. I'm not sure what filename it should be -- is it one passed via the command line?